### PR TITLE
#1731 Docs for 20.0 release

### LINKED ISSUE #1731: Unable to generate documentation on ReadTheDocs
## Expected Behavior / New Feature
We would expect the documentation to be generated on a regular basis, to avoid misunderstandings. 😸 

## Actual Behavior / Motivation for New Feature
The documentation generation process fails at the moment:
[Build #22151302 | Ocelot | Read the Docs](https://readthedocs.org/projects/ocelot/builds/22151302/)

The reason is that Read the Docs has stopped supporting docs generation without the `.readthedocs.yaml` config file:
[Migrate your project to .readthedocs.yaml configuration file v2 — Read the Docs Blog](https://blog.readthedocs.com/migrate-configuration-v2/)

## Solution
A simple solution is given in the PR. By adding a config file to the solution's root folder the docs are generated.
See demo account by @ggnaegi: https://ocelot-test.readthedocs.io/en/bug-fix_documentation_generation/

### PR DESCRIPTION
**Pull request by raman-m**
## Fixes #1731 
- #1731

This is the major update of the documentation for the current [20.0](../releases/tag/20.0.0) release.

## Proposed Changes
  - Reviewed all features, pages in **.rst** files
  - Updated links & features to meet requirements of 20.0 release
  - Updated README.md which looks better and modern now
  - Reviewed development process aka Release Process
  - Added **Future** subsections to say about team plans

I guess, most interesting things to look at are:
- [releaseprocess.rst](../blob/release/20.0-docs/docs/building/releaseprocess.rst)
- [Future](../blob/release/20.0-docs/docs/features/websockets.rst#future) subsections: `git grep 'Future' release/20.0-docs`

### CODE REVIEW

**Comment by raman-m** [comment]
@ggnaegi  FYI
Do you approve this?  Something else?

**Comment by ggnaegi** [comment]
> @ggnaegi  FYI
> The Coveralls.io badge will be [hidden](../blob/release/20.0-docs/README.md?plain=1#L5) on README page until we fix this. :yum: 
> 
> Do you approve this?  Something else?

Yes, until we introduce sonarqube 😊

**Comment by raman-m** [comment]
@ggnaegi Do you approve?
We're very close to releasing v 20.0.1 with this docs.
As a rest of work, I need to add **make.bash** to have ability to build docs on Linux machines. Current **make.bat** allows to build docs on Windows machine only.

**Comment by ggnaegi** [comment]
> @ggnaegi Do you approve?
> We're very close to releasing v 20.0.1 with this docs.
> As a rest of work, I need to add **make.bash** to have ability to build docs on Linux machines. Current **make.bat** allows to build docs on Windows machine only.

@raman-m Sounds good to me. Why "-" instead of "^" ?

**Comment by raman-m** [comment]
>@ggnaegi [commented](#issuecomment-1778649728)
> Sounds good to me. Why "-" instead of "^" ?

- `=` for sections
- `-` for subsections
- `^` for subsubsections

All features have been defined as sections with `=`, and some pages had the next level `^` (subsubsections), which is wrong because the next sublevel after sections is `-` (subsections).

More here: [reST Sections](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections)

**Comment by ggnaegi** [comment]
Ok, reviewing the docs :-)

I would be up for the GraphQL bit, but with https://github.com/ChilliCream/graphql-platform, not graphql.net.

I have some experience with HotChocolate, since I implemented GraphQL for a customer.

this .net server is imho better and Michael (the owner) is a member of the GraphQL Technical Steering Committee.

But that's my opinion.

**Review by ggnaegi — APPROVED** [review]
Great job! I have written two comments, but generally, we should send a positive message to the contributors.

**Comment by raman-m** [comment]
>@ggnaegi [commented](#issuecomment-1778722915)
I would be up for the GraphQL bit, but with https://github.com/ChilliCream/graphql-platform, not graphql.net.

I am not a fun to integrate custom libraries, except direct references to official libs.
Let's discuss GraphQl later, plz!

File: docs/features/qualityofservice.rst
Comment (raman-m) [reply docs/features/qualityofservice.rst:30]: Which one?

Comment (raman-m) [reply docs/features/qualityofservice.rst:30]: If we have a bug in backlog, it doesn't mean we have to remove docs, right?
Make sense?

Comment (raman-m) [reply docs/features/qualityofservice.rst:30]: Or, would you like to mention this bug in docs?
Won't it be strange?

File: docs/building/releaseprocess.rst
Comment (raman-m) [reply docs/building/releaseprocess.rst:50]: Good point! We need to be more pleasant maintainers. 🤣
I would prefer to replace `Do not mention Tom` line by another phrase: "We do follow this dev and release process... bla-bla-bla"

Comment (raman-m) [reply docs/building/releaseprocess.rst:50]: As Tom wrote a lot in docs, I would say `If you need to do something which we won't do, please help yourself!` 🤣

Comment (raman-m) [reply docs/building/releaseprocess.rst:53]: ✔️  Fixed in f9f6d8da5405d46601aacb39308527a51a15dd3b
Are we pleasant now, @ggnaegi ?

**Comment by ggnaegi** [comment]
>  >@ggnaegi [commented](#issuecomment-1778722915)
> I would be up for the GraphQL bit, but with https://github.com/ChilliCream/graphql-platform, not graphql.net.
> 
> I am not a fun to integrate custom libraries, except direct references to official libs.
> Let's discuss GraphQl later, plz!

Mmh. Graphql.NET is not the official implementation by the way. There is no official implementation.

File: docs/features/qualityofservice.rst
Comment (raman-m) [reply docs/features/qualityofservice.rst:30]: I have an idea to mark this feature by special label and redirect users to look at current issues? Souds good? Sounds like a fix of this code review issue?
For example we have [![label: Websockets][~websockets]](../labels/Websockets) label for **Websockets** feature.

[~websockets]: https://img.shields.io/badge/-Websockets-BFDADC.svg

Comment (raman-m) [reply docs/features/qualityofservice.rst]: I found it! It is https://github.com/ThreeMammals/Ocelot/issues/1550 

@ggnaegi  I've added special QoS label, and added the reference to QoS footnote in the doc
Fixed in https://github.com/ThreeMammals/Ocelot/commit/88fbd1d396a8deccea04e7b1b6cac03db35f1ecf

**Comment by raman-m** [comment]
> @ggnaegi left [a code review comment](#pullrequestreview-1696633710):
> Great job!

Thanks!

> I have written two comments, but we should send a positive message to the contributors.

I've fixed issues! Please, approve, if no other issues.

**Comment by raman-m** [comment]
@RaynaldM  Hey, Ray!
One more approval please!

**Review by ggnaegi — APPROVED** [review]
Ok for me, thanks!

**Comment by raman-m** [comment]
@RaynaldM Good morning, France! 😊

**Pull request merged**
